### DISCUSSION TITLE
Customize the Compass?

### DISCUSSION THREAD

**Discussion by Uranbold** [maplibre/maplibre-native#2223] [Q&A]
I wanna customize the Compass Icon? Is there any plan to make it possible?

**Comment by louwers** [comment]
What platform?

**Comment by Mulczas** [comment]
You can make your own compass component and disable built-in by setting "isCompassEnabled" in "uiSettings: to false. Then add two camera listeners: addOnCameraDidChangeListener and addOnCameraIsChangingListener. In both listeners read bearing from cameraPosition and rotate your component by given value (with negative value).
